### PR TITLE
Support IOPS parameter for IO1/IO2 volumes

### PR DESCRIPTION
**Pull request by torredil**
**What is this PR about? / Why do we need it?**
- Support IOPS parameter for IO1/IO2 volumes.
- Support IOPSPerGB for GP3 volumes.
- Address #1364.

**What testing is done?** 

- `make test`
- CI

```
StorageClass
---------------------------
apiVersion: storage.k8s.io/v1
kind: StorageClass
metadata:
  name: ebs-sc
provisioner: ebs.csi.aws.com
volumeBindingMode: WaitForFirstConsumer
parameters:
  csi.storage.k8s.io/fstype: ext4
  IOPS: "100"
  type: io2
---------------------------

**Old behavior**

$ kubectl describe pvc

Warning  ProvisioningFailed    2s (x3 over 5s)  ebs.csi.aws.com_ebs-csi-controller-756d5594dd-vgzsj_728a3701-f044-4021-bfaf-316189032728  failed to provision volume with StorageClass "ebs-sc": rpc error: code = Internal desc = Could not create volume "pvc-6ebcdc8f-0ab7-4e9a-9ee4-477c7cbb7d87": invalid combination of volume size 4 GB and iopsPerGB 0: the resulting IOPS 0 is too low for AWS, it must be at least 100

**New behavior**

$ kubectl describe pvc

Normal   ProvisioningSucceeded  3s                 ebs.csi.aws.com_ebs-csi-controller-58f67d8bfb-fm2m7_9ca56fcb-bcc9-4ba8-a2e3-872db5595932  Successfully provisioned volume pvc-36e7c4ac-00cd-4340-9517-f3b5bcac6da8
```
-  Failure mode: specifying `IOPS` and `IOPSPerGB` StorageClass parameters.
```
StorageClass
---------------------------
apiVersion: storage.k8s.io/v1
kind: StorageClass
metadata:
  name: ebs-sc
provisioner: ebs.csi.aws.com
volumeBindingMode: WaitForFirstConsumer
parameters:
  csi.storage.k8s.io/fstype: ext4
  IOPS: "100"
  IOPSPerGB: "100"
  type: io2
---------------------------

$ kubectl describe pvc

Warning  ProvisioningFailed    2s (x2 over 3s)  ebs.csi.aws.com_ebs-csi-controller-58f67d8bfb-fm2m7_9ca56fcb-bcc9-4ba8-a2e3-872db5595932  failed to provision volume with StorageClass "ebs-sc": rpc error: code = Internal desc = Could not create volume "pvc-2c35f4c6-cc22-48c1-8d40-3a42645b7ee3": invalid StorageClass paramaters; specify either IOPS or IOPSPerGb, not both
```

*Signed-off-by: Eddie Torres <torredil@amazon.com>*

### CODE REVIEW

File: pkg/cloud/cloud.go [hunk 1]
```diff
@@ -283,29 +283,37 @@ func (c *cloud) CreateDisk(ctx context.Context, volumeName string, diskOptions *
...
+				return nil, err
+			}
 		}
-	case VolumeTypeGP3:
+	case VolumeTypeGP3, "":
 		createType = diskOptions.VolumeType
 		iops = int64(diskOptions.IOPS)
 		throughput = int64(diskOptions.Throughput)
-	case "":
-		createType = DefaultVolumeType
```
Comment (ConnorJC3) [inline pkg/cloud/cloud.go:306]: If you're going to remove this line you should also remove the `DefaultVolumeType` constant

Comment (torredil) [reply pkg/cloud/cloud.go:306]: Done.

**Comment by torredil** [comment]
@ConnorJC3 Good point, refactored this to clean up the duplication and unify iops handling across volume types.

File: pkg/cloud/cloud.go [hunk 2]
```diff
@@ -276,40 +277,57 @@ func newEC2Cloud(region string, awsSdkDebugLog bool) (Cloud, error) {
...
+		minIops = gp3MinTotalIOPS
+		maxIopsPerGb = gp3MaxIOPSPerGB
 		throughput = int64(diskOptions.Throughput)
 	default:
 		return nil, fmt.Errorf("invalid AWS VolumeType %q", diskOptions.VolumeType)
 	}
 
+	if maxIops > 0 {
+		if diskOptions.IOPS > 0 {
+			iops = int64(diskOptions.IOPS)
```
Comment (rdpsin) [inline pkg/cloud/cloud.go:322]: What happens if I specify 6000 IOPS on a 10GiB io2 volume? The [doc](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ebs-volume-types.html) states that the maximum ratio is 500:1, so the maximum allowed IOPS will be 5000. Do we need to explicitly handle the case or will CreateVolume handle that?

Comment (torredil) [reply pkg/cloud/cloud.go]: It should be explicitly handled, newest revision accounts for this.

**Pull request merged**